# This patch file was generated by NetBeans IDE
# Following Index: paths are relative to: lifedesk/sites/admin.lifedesks.org/modules/mass_contact
# This patch can be applied using context Tools: Patch action on respective folder.
# It uses platform neutral UTF-8 encoding and \n newlines.
# Above lines and this line are ignored by the patching process.
Index: mass_contact.info
--- mass_contact.info Base (BASE)
+++ mass_contact.info Locally Modified (Based On LOCAL)
@@ -11,3 +11,5 @@
 project = "mass_contact"
 datestamp = "1273608907"
 
+; Added by D. Shorthouse to permit multisite mass mail
+dependencies[] = db_switch
\ No newline at end of file
Index: mass_contact.module
--- mass_contact.module Base (BASE)
+++ mass_contact.module Locally Modified (Based On LOCAL)
@@ -72,10 +72,12 @@
     'send mass contact e-mails',
   );
 
+  if(db_table_exists('mass_contact')) {
   $result = db_query('SELECT category FROM {mass_contact}');
   while ($category = db_fetch_object($result)) {
     $permissions[] = 'send to users in the '. $category->category .' category';
   }
+  }
 
   return $permissions;
 }  //  End of mass_contact_perm().
@@ -1377,6 +1379,9 @@
     $roles = explode(',', $contact->recipients);
     foreach ($roles as $r) {
       if ($r == 2) { // all users
+      $recipientouta = _mass_contact_db_switch();
+// Commented out by D. Shorthouse and instead replaced with above function call
+/*
         $recipients = db_query("SELECT name, mail, uid FROM {users} WHERE status <> %d AND uid > %d", 0, 0);
         while ($obj = db_fetch_object($recipients)) {
           $rname = $obj->name;
@@ -1388,8 +1393,12 @@
           }
         }
         break;
+ */
       }
       else {
+        $recipientouta = _mass_contact_db_switch($r);
+// Commented out by D. Shorthouse and instead replaced with above function call
+/*
         // Get from users_roles, then role -> user.
         $uids = db_query("SELECT ur.uid FROM {users_roles} ur LEFT JOIN {users} u ON ur.uid = u.uid WHERE ur.rid = %d AND u.status <> %d", $r, 0);
         while ($obj = db_fetch_object($uids))  {
@@ -1409,6 +1418,7 @@
             }
           }
         }
+ */
       }
     }
 
@@ -1824,3 +1834,49 @@
 
   return $message;
 }  //  End of _mass_contact_mail().
+
+/*
+ * Function added by D. Shorthouse to accommodate multisite env't
+ */
+function _mass_contact_db_switch($r='') {
+
+	$recipientouta = array();
+
+	$query = db_query("SELECT shortname FROM {drupal_site}");
+
+	while ($drupal_site = db_fetch_object($query)) {
+	  $switch_db = db_switch($drupal_site->shortname);
+
+	  //get the users
+
+	  // all users
+	  if(!$r) {
+        $recipients = db_query("SELECT name, mail, uid FROM {users} WHERE status <> %d AND uid > %d", 0, 1);
+        while ($obj = db_fetch_object($recipients)) {
+          $rname = $obj->name;
+          $rmail = $obj->mail;
+          $ruid = $obj->uid;
+          if ($rname != $user->name) {
+            $recipientouta[$rname] = $rmail;
+          }
+        }
+        break;
+	  }
+	  else {
+        $uids = db_query("SELECT ur.uid FROM {users_roles} ur LEFT JOIN {users} u ON ur.uid = u.uid WHERE ur.rid = %d AND u.status <> %d", $r, 0);
+        while ($obj = db_fetch_object($uids))  {
+          $ruid = $obj->uid;
+          $userobj = db_fetch_object(db_query("SELECT name, mail FROM {users} WHERE uid = %d", $ruid));
+          $rname = $userobj->name;
+          $rmail = $userobj->mail;
+          if ($rname != $user->name) {
+            $recipientouta[$rname] = $rmail;
+          }
+        }
+	  }
+
+	  $switch_db = db_switch();
+	}
+
+	return $recipientouta;
+}
\ No newline at end of file
